modelzoo/DLRM/train.py

```diff
@@ -224,50 +190,400 @@ def prediction(self):
             net = tf.cast(net, dtype=tf.bfloat16)
         mlp_top_scope = tf.variable_scope(
             'mlp_top_layer',
-            partitioner=self.dense_layer_partitioner,
+            partitioner=self._dense_layer_partitioner,
             reuse=tf.AUTO_REUSE)
         with mlp_top_scope.keep_weights(dtype=tf.float32) if self.bf16 \
             else mlp_top_scope:
-            for layer_id, num_hidden_units in enumerate(self.mlp_top):
+            for layer_id, num_hidden_units in enumerate(self._mlp_top):
                 with tf.variable_scope(
-                        "mlp_top_hiddenlayer_%d" % layer_id,
+                        'mlp_top_hiddenlayer_%d' % layer_id,
                         reuse=tf.AUTO_REUSE) as mlp_top_hidden_layer_scope:
                     net = tf.layers.dense(net,
                                           units=num_hidden_units,
                                           activation=tf.nn.relu,
                                           name=mlp_top_hidden_layer_scope)
 
-                add_layer_summary(net, mlp_top_hidden_layer_scope.name)
+                self._add_layer_summary(net, mlp_top_hidden_layer_scope.name)
 
         if self.bf16:
             net = tf.cast(net, dtype=tf.float32)
 
-        with tf.variable_scope("logits", reuse=tf.AUTO_REUSE) as logits_scope:
+        with tf.variable_scope('logits', reuse=tf.AUTO_REUSE) as logits_scope:
             net = tf.layers.dense(net,
                                   units=1,
                                   activation=None,
                                   name=logits_scope)
             net = tf.math.sigmoid(net)
 
-            add_layer_summary(net, logits_scope.name)
+            self._add_layer_summary(net, logits_scope.name)
 
-        return net
+        self._predict = net
```

Comment:
self.output is needed. please follow template

Reply:
revised